appscale/tools/agents/azure_agent.py

```diff
@@ -454,6 +532,19 @@ def create_or_update_vm_scale_sets(self, count, parameters, subnet):
     verbose = parameters[self.PARAM_VERBOSE]
     random_resource_name = Haikunator().haikunate()
 
+    # While autoscaling, look through existing scale sets to check if they have
+    # capacity to hold more vms. If they do, update the scale sets with additional
+    # vms. If not, then create a new scale set for them.
+    is_autoscale = parameters['autoscale_agent']
+    num_instances_to_add = count
+    if is_autoscale in ['True', True]:
+      instances_added = self.add_instances_to_existing_ss(
+        num_instances_to_add, parameters)
+      if num_instances_to_add > instances_added:
+        count = count - instances_added
```

Comment:
I would recommend modifying `num_instances_to_add` instead of `count` just in case someone later decides to edit this function and assumes (from the docstring) that count always refers to the number the client asked for.